zerver/models.py

```diff
@@ -1400,6 +1399,13 @@ class UserBaseSettings(models.Model):
         presence_enabled=bool,
     )
 
+    notification_setting_types = {
+        **notification_settings_legacy
+    }  # Add new notifications settings here.
```

Comment:
I think having this would keep us more organized and function names like [do_change_notification_settings](https://github.com/zulip/zulip/blob/master/zerver/lib/actions.py#L5018) and some instances where we use this make more sense with it named as `notification_setting_types` instead of `notification_settings_legacy`.